### ISSUE TITLE
[BUG] S3 Artifact 

### ISSUE THREAD

**Issue by lokeish** [mlflow/mlflow#9377]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

- Client: 1.x.y
- Tracking server: 1.x.y

### System information

Ubuntu 22.04

### Describe the problem

Hi Team, 

There is a bug in the log_artifact when trying to store in was s3 bucket(Not Minio), the bug is basically on the bucket region. By default, mlflow is trying to store in us-east-1,  but in my case, the bucket is present in ap-southeast-2.  So I have gone through the source code of mlflow, and found that there is no region handling for the s3 client.

It would be great if the team fixes this issue, Just region addition is required while creating the s3 client.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@lokeish Can you try specifying the region in `~/.aws/config`?

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html

**Comment by lokeish** [comment]
> @lokeish Can you try specifying the region in `~/.aws/config`? https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html

The region is ap-southeast-2

**Comment by harupy** [comment]
so you have already tried ` ~/.aws/config`?

**Comment by harupy** [comment]
@lokeish Can you upload a file without any issues using only boto3?

**Comment by lokeish** [comment]
> ~/.aws/config

I have verified from my end,  in the ~/.aws/config the default region is nothing, but here is a catch, i tried with the normal boto3 python program to upload to ap-southeast-2 region bucket, and the upload is successful but the same thing is getting failed from the mlflow side.

**Comment by lokeish** [comment]
> @lokeish Can you upload a file without any issues using only boto3?

``` 
import boto3

# Replace these with your own credentials and bucket name
aws_access_key_id = 'xxx'
aws_secret_access_key = 'xxx'
bucket_name = 'doc-mlflow'

local_file = '/home/lokeish/Documents/exp_details.json'  # Path to the local file you want to upload
destination_key = 'uploaded-file_11.txt'  


# Create an S3 client
s3_client = boto3.client(
    's3',
    aws_access_key_id=aws_access_key_id,
    aws_secret_access_key=aws_secret_access_key
)


def upload_file_with_put_object(bucket_name, local_file, destination_key):
    try:
        with open(local_file, 'rb') as f:
            s3_client.put_object(Bucket=bucket_name, Key=destination_key, Body=f)
        print(f"File '{local_file}' uploaded as '{destination_key}' to '{bucket_name}'")
    except Exception as e:
        print("Error:", e)


def list_files_in_bucket(bucket_name):
    response = s3_client.list_objects_v2(Bucket=bucket_name)

    if 'Contents' in response:
        for obj in response['Contents']:
            print(obj['Key'])
    else:
        print("Bucket is empty or no objects found.")

# List files in the bucket
list_files_in_bucket(bucket_name)
upload_file_with_put_object(bucket_name, local_file, destination_key)
```

**Comment by harupy** [comment]
Interesting. This is how MLflow initializes the boto3 client:

https://github.com/mlflow/mlflow/blob/304ca3d2598dee681b311ad6add85c7b51f482fe/mlflow/store/artifact/s3_artifact_repo.py#L61-L69

**Comment by lokeish** [comment]
> 

Yes exactly, I have gone through the code, and it should provide an option to take the bucket region, because ~/.aws/config is global to the host, and in my case, my program is trying to deal with different region buckets.

**Comment by lokeish** [comment]
Hi [harupy](https://github.com/harupy),

I have deployed the mlflow server using the docker container where I have provided the region details in env files, but still, somehow mlflow couldn't able to pick the details from region-related env value.

**Comment by harupy** [comment]
> my program is trying to deal with different region buckets.

Does setting `AWS_DEFAULT_REGION` work?

**Comment by piyushdaftary** [comment]
This issue seems to have addressed by PR : https://github.com/mlflow/mlflow/pull/9498
